### ISSUE TITLE
importing crypt module in Windows causes "maximum recursion depth exceeded"

### ISSUE THREAD

**Issue by matthchr** [pyinstaller/pyinstaller#2875]
The below error occurs when bundling the application (actually, during the Analysis process I think).

I am running on the following platform:
```
285 INFO: PyInstaller: 3.3
285 INFO: Python: 3.6.2
286 INFO: Platform: Windows-10-10.0.15063-SP0
288 INFO: UPX is not available.
```
I believe it also repo's on Python 3.4.3.

I can repro this issue with a simple 1 line script:
```
import crypt
```

If you run `pyinstaller main.py` on a script containing the single line above, you get the following error:
```
...
  File "c:\users\matthchr\pycharmprojects\pyinstallertest\venv\lib\site-packages\PyInstaller\depend\analysis.py", line 315, in _find_module_path
    fullname, module_name, search_dirs)
  File "c:\users\matthchr\pycharmprojects\pyinstallertest\venv\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 2934, in _find_module_path
    module_name)
  File "<frozen importlib._bootstrap_external>", line 1223, in find_loader
  File "<frozen importlib._bootstrap_external>", line 1268, in find_spec
  File "<frozen importlib._bootstrap_external>", line 60, in _path_join
RecursionError: maximum recursion depth exceeded
```
[Full error log here](https://github.com/pyinstaller/pyinstaller/files/1334495/pyinstaller.crypt.error.txt)

It turns out, that since crypt only works on Linux, that there is some issue with importing it in Windows, even though it actually works just fine as long as you don't actually call it in Windows.

Editing the spec file to explicitly ignore `crypt` or just `_crypt` solves the issue. 

It would be nice to have a clearer error code though.

**Comment by htgoebel** [comment]
Thanks for reporting. This looks like a bug we need to track down. Can you please run `pyinstaller --log =TRACE` and search for some insight there. It will produce a *huge* amount of log-entries, thus searching for `crypt` or `_crypt` should bring you into the relevant area.

On Linux, `_crypt` is a .so. What is it on Windows?

> It turns out, that since crypt only works on Linux

Does it? [The documentation](https://docs.python.org/3/library/crypt.html) does not say so.

**Comment by matthchr** [comment]
[This stackoverflow post](https://stackoverflow.com/questions/19877867/using-the-crypt-module-in-windows) points out that the crypt docs are under the "unix specific services" section - so yes I think it is unix specific.

`_crypt` doesn't exist in windows (as far as I can tell) -- and that's definitely where the stack overflow in PyInstaller is coming from since if I explicitly ignore `_crypt` everything works.

As for the Trace log, I've attached the file [here](https://github.com/pyinstaller/pyinstaller/files/1338109/trace.txt)

Important snippet is:
```
11725 TRACE: safe_import_module -> None (ImportError("No module named '_crypt'",)) 
11725 TRACE: raise ImportError: No module named '_crypt'
11725 TRACE: SWIG import candidate (name='_crypt', caller=SourceModule('crypt', 'd:\\python\\3.6\\Lib\\crypt.py'), level=0) 
11725 TRACE: ImportError: 'No module named _crypt'
11725 TRACE: addNode MissingModule('_crypt',)


11726 TRACE: _import_hook '_crypt' SourceModule('crypt', 'd:\\python\\3.6\\Lib\\crypt.py') 
SourceModule('crypt', 'd:\\python\\3.6\\Lib\\crypt.py') 0
11728 TRACE: safe_import_module -> None (ImportError("No module named '_crypt'",)) 
11728 TRACE: raise ImportError: No module named '_crypt'
```

It looks like PyInstaller is getting stuck in a loop trying to resolve `_crypt`, since it basically keeps repeating the same stuff over and over.

**Comment by sflyme** [comment]
I found the same error.   My trace log see that: deadlock  loop (_import_hook  '_ssl') 
python3.6.3
[trace.log](https://github.com/pyinstaller/pyinstaller/files/1397262/trace.log)

pyinstaller3.3

**Comment by htgoebel** [comment]
Maybe https://github.com/pyinstaller/pyinstaller/issues/2911#issuecomment-336422915 is the cause for this issue, too?! To track down the issue, please revert commit 711e9e77c93a979a63648ba05f725b30dbb7c3cc and report back whether this helped. Thanks,

**Comment by htgoebel** [comment]
@matthchr I'm still waiting for an answer. Please revert commit 711e9e77c93a979a63648ba05f725b30dbb7c3cc and report back whether this solves the issue. 

If this does not solve the issue, please try xoviat 's pull-request:
```
git clone https://github.com/pyinstaller/pyinstaller.git
cd pyinstaller
git pull origin +refs/pull/3024/merge
pip install .
```
Thanks.

**Comment by matthchr** [comment]
Sorry for the slow response @htgoebel - I was able to work around this issue by turning on verbose, finding the import it was stuck on and then manually excluding it via the `excludes` parameter to `Analysis`. For me, it was happening for `_crypt` on Windows and also for `_decimal` on Linux, so the issue isn't specific to Windows.

This seems to primarily be a reporting/debugability problem.

**Comment by htgoebel** [comment]
The issue should be solved, not worked around, shouldn't it? So please help tracking this bug down. I can not reproduce the issue here. Thanks.

**Comment by htgoebel** [comment]
This should be solved now by #3061. Please try the current development version. If the problem still persists, please drop a note and reopen the ticket. Thanks.
